src/client/inputhandler.h

```diff
@@ -60,32 +71,59 @@ class MyEventReceiver : public IEventReceiver
 #endif
 		// handle mouse events
 		if (event.EventType == irr::EET_MOUSE_INPUT_EVENT) {
-			if (noMenuActive() == false) {
-				left_active = false;
-				middle_active = false;
-				right_active = false;
-			} else {
-				left_active = event.MouseInput.isLeftPressed();
-				middle_active = event.MouseInput.isMiddlePressed();
-				right_active = event.MouseInput.isRightPressed();
-
-				if (event.MouseInput.Event == EMIE_LMOUSE_PRESSED_DOWN) {
-					leftclicked = true;
-				}
-				if (event.MouseInput.Event == EMIE_RMOUSE_PRESSED_DOWN) {
-					rightclicked = true;
-				}
-				if (event.MouseInput.Event == EMIE_LMOUSE_LEFT_UP) {
-					leftreleased = true;
-				}
-				if (event.MouseInput.Event == EMIE_RMOUSE_LEFT_UP) {
-					rightreleased = true;
-				}
+			// handle mouse buttons
+			// left
+			if (event.MouseInput.Event == EMIE_LMOUSE_PRESSED_DOWN) {
+			     	if (keyIsDown["KEY_LBUTTON"]) {
+					keyPressed.set("KEY_LBUTTON");
+			     	}
+				keyDown.set("KEY_LBUTTON");
```

Comment:
@Zeno- I think we should let so make another pull request with a rework of the KeyList Class.